### PR TITLE
Introduce EA pattern for TypeScript for navbars.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/52632

Draft PR until https://github.com/dotnet/roslyn/pull/52530 goes in.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @uniqueiniquity  and @minestarks  No rush on this.

**Comment by CyrusNajmabadi** [comment]
@sharwell  @dotnet/roslyn-ide  please review.

**Comment by CyrusNajmabadi** [comment]
@uniqueiniquity @minestarks the order of operations will be:

1. this PR goes in. Roslyn continues to defer to your existing INavigationBarItemService impl
2. you get our bits. you then convert your INavigationBarItemService impl to instead impl IVSTypeScriptNavigationBarItemService. This should be relatively painless. The major difference are that the apis use immutbale arrays instead of mutable lists, and are async in a few more places.
3. Remember to stop implementing INavigationBarItemService . Our system will pick up your IVSTypeScriptNavigationBarItemService.
4. Once you get merged, let us know. We will then remove our bridge interface (called INavigationBarItemServiceRenameOnceTypeScriptMovesToExternalAccess).

File: src/EditorFeatures/Core/Extensibility/NavigationBar/INavigationBarItemService.cs
Comment (allisonchou) [inline src/EditorFeatures/Core/Extensibility/NavigationBar/INavigationBarItemService.cs:14]: Same comment as the other PR, I think it would be helpful if the comment here was expanded. But if TS plans to adopt these changes quickly then maybe it doesn't matter too much.

**Review by allisonchou — APPROVED** [review]

**Pull request merged**
